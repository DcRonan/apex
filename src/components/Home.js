import React from 'react';
import Footer from './Footer';
import Routes from '../routes';

const Home = () => (
  <>
    <main>
      <Routes />
    </main>
    <Footer />
  </>
);

export default Home;
