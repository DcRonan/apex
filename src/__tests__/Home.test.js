import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Home from '../components/AddSport';
import store from '../store';

describe('checks if Home component renders correctly', () => {
  test('should renders correct text', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>,
    );
    const text = screen.getByText('Name of sport');
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Home />
          </BrowserRouter>
          ,
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
