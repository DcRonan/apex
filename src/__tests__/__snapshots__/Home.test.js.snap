// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Home component renders correctly should render correctly 1`] = `
Array [
  <div
    className="text-center h-full"
  >
    <div
      className="p-4 bg-lightblue text-white font-nimbuslight w-full"
    >
      <h1
        className="text-2xl"
      >
        Add sport
      </h1>
    </div>
    <div
      className="p-4 border-b-2"
    >
      <div>
        <h3
          className="text-base text-gray-500 font-nimbusbold"
        >
          Name of sport
        </h3>
      </div>
    </div>
    <div
      className="flex flex-col items-center justify-center space-y-4"
    >
      <div
        className="flex flex-col items-center justify-center py-10 font-nimbuslight"
      >
        <section
          className="hidden w-3/4"
        >
          <svg
            className="CircularProgressbar "
            data-test-id="CircularProgressbar"
            style={Object {}}
            viewBox="0 0 100 100"
          >
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "289.02652413026095px",
                }
              }
            />
            <text
              className="CircularProgressbar-text"
              style={Object {}}
              x={50}
              y={50}
            >
              0 h
            </text>
          </svg>
        </section>
        <section
          className="hidden w-3/4"
        >
          <svg
            className="CircularProgressbar "
            data-test-id="CircularProgressbar"
            style={Object {}}
            viewBox="0 0 100 100"
          >
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "289.02652413026095px",
                }
              }
            />
            <text
              className="CircularProgressbar-text"
              style={Object {}}
              x={50}
              y={50}
            >
              0 mins
            </text>
          </svg>
        </section>
        <section
          className="hidden w-3/4"
        >
          <svg
            className="CircularProgressbar "
            data-test-id="CircularProgressbar"
            style={Object {}}
            viewBox="0 0 100 100"
          >
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "289.02652413026095px",
                }
              }
            />
            <text
              className="CircularProgressbar-text"
              style={Object {}}
              x={50}
              y={50}
            >
              0 s
            </text>
          </svg>
        </section>
        <section
          className="hidden w-3/4"
        >
          <svg
            className="CircularProgressbar "
            data-test-id="CircularProgressbar"
            style={Object {}}
            viewBox="0 0 100 100"
          >
            <path
              className="CircularProgressbar-trail"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "0px",
                }
              }
            />
            <path
              className="CircularProgressbar-path"
              d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
              fillOpacity={0}
              strokeWidth={8}
              style={
                Object {
                  "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                  "strokeDashoffset": "289.02652413026095px",
                }
              }
            />
            <text
              className="CircularProgressbar-text"
              style={Object {}}
              x={50}
              y={50}
            >
              0 miles
            </text>
          </svg>
        </section>
      </div>
      <form
        className="space-y-4 absolute bottom-20 font-nimbuslight"
        onSubmit={[Function]}
      >
        <input
          name="name"
          onChange={[Function]}
          placeholder="name"
          type="text"
          value=""
        />
        <input
          className="hidden"
          name="minutes"
          onChange={[Function]}
          placeholder="minutes"
          type="number"
          value={0}
        />
        <input
          className="hidden"
          name="seconds"
          onChange={[Function]}
          placeholder="seconds"
          type="number"
          value={0}
        />
        <input
          className="hidden"
          name="hours"
          onChange={[Function]}
          placeholder="hours"
          type="number"
          value={0}
        />
        <input
          className="hidden"
          name="distance"
          onChange={[Function]}
          placeholder="distance"
          type="number"
          value={0}
        />
        <textarea
          className="hidden"
          cols="20"
          name="notes"
          onChange={[Function]}
          placeholder="notes"
          required={true}
          rows="6"
          value=""
        />
        <div
          className="bg-gray-100 rounded-lg p-2"
        >
          <button
            className="p-2 bg-lightgreen text-white w-full rounded-lg"
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
        <div
          className="hidden"
        >
          <button
            className="p-3 rounded-lg text-gray-600 w-full"
            onClick={[Function]}
            type="button"
          >
            Prev
          </button>
          <button
            className="p-3 rounded-lg bg-lightgreen text-white w-full"
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
        <div
          className="hidden"
        >
          <button
            className="p-3 text-grey-600 w-full text-gray-600"
            onClick={[Function]}
            type="button"
          >
            Prev
          </button>
          <button
            className="p-3 bg-lightgreen text-white w-full rounded-lg"
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
        <div
          className="hidden"
        >
          <button
            className="p-3 rounded-lg text-gray-600 w-full"
            onClick={[Function]}
            type="button"
          >
            Prev
          </button>
          <button
            className="p-3 bg-lightgreen text-white w-full rounded-lg"
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
        <div
          className="hidden"
        >
          <button
            className="p-3 text-gray-600 w-full"
            onClick={[Function]}
            type="button"
          >
            Prev
          </button>
          <button
            className="p-3 bg-lightgreen text-white w-full rounded-lg"
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
        <div
          className="hidden"
        >
          <button
            className="p-3 text-gray-600 w-full"
            onClick={[Function]}
            type="button"
          >
            Prev
          </button>
          <button
            className="p-3 bg-lightgreen text-white w-full rounded-lg"
            onClick={[Function]}
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  ",",
]
`;
